### PR TITLE
Migrate to 1ES templates

### PR DESCRIPTION
**Pull request by idavis**

### CODE REVIEW

File: .ado/publish.yml
Comment (DmitryVasilevsky) [inline .ado/publish.yml:370]: Sounds silly, but I suggest moving this to pre-approval if possible. I think we should have as few steps for actual publishing as possible so that we have fewer failures after approval. I don't see why this should fail, but it would be beneficial to catch failures in the part that executes often.
(Feel free to ignore.)

Comment (idavis) [reply .ado/publish.yml:370]: This can't be moved as we are on a fresh image for the job and no data from the previous jobs is available except that which is fed as a pipeline artifact.

**Review by billti — APPROVED** [review]

**Review by swernli — APPROVED** [review]

**Pull request merged**
